File: app/src/main/java/io/oldering/tvfoot/red/di/module/MatchActivityModule.java
package io.oldering.tvfoot.red.di.module;

import android.app.Activity;
import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import io.oldering.tvfoot.red.di.component.MatchActivitySubComponent;
import io.oldering.tvfoot.red.match.MatchActivity;

@Module(subcomponents = MatchActivitySubComponent.class) //
public abstract class MatchActivityModule {
  @Binds @IntoMap @ActivityKey(MatchActivity.class) //
  abstract AndroidInjector.Factory<? extends Activity> bindMatchActivityInjectorFactory(
      MatchActivitySubComponent.Builder builder);
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/module/MatchActivityModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import android.app.Activity
import com.benoitquenaudon.tvfoot.red.app.domain.match.MatchActivity
import dagger.Binds
import dagger.Module

@Module
abstract class MatchActivityModule {
  @Binds abstract fun provideActivity(activity: MatchActivity): Activity
}